lib/plugins/aws/utils/iam.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+exports.getDeploymentRole = (serverless) => {
```

Comment:
I believe right way to approach that is to:

1. Introduce `provider.customDeploymentRole` (where `provider` is instance of `AwsProvider` and update all `service.provider.cfnRole` occurences to rely on `provider.customDeploymentRole` instead

2. Introduce `provider.customExecutionRole` and update all points where we access it, so this property is accessed.

Also this is a refactor that we can do in other PR, and then in this PR just update initial resolution of `provider.customDeploymentRole` and `provider.customExecutionRole`.

It'll be great to split this work into multiple PR's that way. There's less chance risk we break something and history of changes in repo is much better (note that one PR ends as single commit)

Reply:
@medikoo makes sense re deployment role.

Not 100% sure how you think it should work for execution role when function params are in the mix?

Reply:
> Not 100% sure how you think it should work for execution role when function params are in the mix?

Good point, I forgot it can be customized per function.

In light of that I propose to introduce `provider.resolveFunctionIamRoleArn` method (we already have e.g. [`  resolveFunctionIamRoleResourceName`](https://github.com/serverless/serverless/blob/2b7568a960c88dda8ab2bbe1b6c8dd238fa78a51/lib/plugins/aws/provider.js#L1537))